iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -815,8 +815,8 @@ private void assertFileFormat(String format) {
   }
 
   private void setCommonHmsTablePropertiesForIceberg(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
-    // If the table is not managed by Hive catalog then the location should be set
-    if (!Catalogs.hiveCatalog(conf, catalogProperties)) {
+    // If the table is not managed by Hive or Hadoop catalog, then the location should be set
+    if (!Catalogs.hiveCatalog(conf, catalogProperties) && !Catalogs.hadoopCatalog(conf, catalogProperties)) {
```

Comment:
I don't get this change. I still can create Hadoop catalog tables without setting the location.
```
set iceberg.catalog.ice01.type=hadoop;
set iceberg.catalog.ice01.warehouse=hdfs://127.0.0.1:8020/tmp;

CREATE EXTERNAL TABLE orders (orderid INT, quantity INT, itemid INT, tradets TIMESTAMP) PARTITIONED BY (p1 STRING, p2 STRING) STORED BY ICEBERG STORED AS ORC;
```

Reply:
````
 org.apache.hadoop.hive.ql.exec.tez.TezRuntimeException: Vertex failed, vertexName=Map 1, vertexId=vertex_1752223623655_0001_3_00, diagnostics=[Vertex vertex_1752223623655_0001_3_00 [Map 1] killed/failed due to:ROOT_INPUT_INIT_FAILURE, Vertex Input: orders initializer failed, vertex=vertex_1752223623655_0001_3_00 [Map 1], java.io.IOException: cannot find dir = /tmp/default/orders in pathToPartitionInfo: [file:/Users/dkuzmenko/projects/hive/upstream/hive/itests/qtest-iceberg/target/localfs/warehouse/orders]
````
table dir should be pre-created